### ISSUE TITLE
[FR] support default size (scalar) in torch.randint

### ISSUE THREAD

**Issue by SsnL** [pytorch/pytorch#18315]
## Motivation

This will be useful for people implementing stochastic algorithms and want the script to respect pytorch seed. np also has this.

**Comment by iArunava** [comment]
I want to take this. Can you point me to some files?
And just to make sure, I understood it correctly
```
>>> import numpy as np
>>> np.random.randint(10)
8
```
while in torch
```
>>> import torch
>>> torch.randint(1, 10, (1, 1))
tensor([[8]])
```
so, it needs to be changed to something like
```
>>> torch.randint(1, 10)
tensor(8)
```
or ?
```
>>>torch.randint(10)
tensor(8)
```

**Comment by colesbury** [comment]
@iArunava, `torch.randint(1, 10, (1, 1))` needs to continue to work. Otherwise, yes  `torch.randint(1, 10)` and `torch.randint(10)` should behave as you suggest.

Note that this is equivalent to writing `torch.randint(1, 10, ())` and `torch.randint(10, ())` which currently work. 

The documentation also needs to be updated:

https://pytorch.org/docs/stable/torch.html#torch.randint

`size` should have a default value of `None`

**Comment by ezyang** [comment]
It might be tricky to get this working C++ side "properly", I don't know if our overload machinery can handle this correctly. Worth a try, I guess.

**Comment by iArunava** [comment]
I am very willing to take this up, but I can't as I can't build pytorch in `develop` mode, if someone is willing to assist it will be very helpful. Here: https://discuss.pytorch.org/t/building-from-source-keeps-failing-ubuntu-18-04-02-lts-no-gpu/41637 Otherwise, I first need to figure out what is going wrong, as I won't be able to test that the changes I make are working or not.

**Comment by iArunava** [comment]
I have successfully build the pytorch code from source. So, I will take this up. Some pointers to files can give me a headstart. Thanks :)

**Comment by colesbury** [comment]
@iArunava these files are relevant:

https://github.com/pytorch/pytorch/tree/master/aten/src/ATen/native/README.md

https://github.com/pytorch/pytorch/blob/a7095b355ee76fcfcf2dcf0600e15b42c92d3a1c/aten/src/ATen/native/native_functions.yaml#L1744-L1764

https://github.com/pytorch/pytorch/blob/a7095b355ee76fcfcf2dcf0600e15b42c92d3a1c/aten/src/ATen/native/TensorFactories.cpp#L297

**Comment by iArunava** [comment]
Thanks @colesbury . I was going through `TensorFactories.cpp`, will take a look at `native_functions.yaml` and start making the PR.

**Comment by iArunava** [comment]
Hi, What's the best and quick way to recompile the code and test after I make the changes to cpp files?

Also, where exactly is `at::empty` and `random_` defined? The core function which this calls.
https://github.com/pytorch/pytorch/blob/a7095b355ee76fcfcf2dcf0600e15b42c92d3a1c/aten/src/ATen/native/TensorFactories.cpp#L325-L326

**Comment by soumith** [comment]
@iArunava for recompile / test, see the [contributing guide](https://github.com/pytorch/pytorch/blob/master/CONTRIBUTING.md) which has lots of tips
